aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -9,51 +10,73 @@
 namespace at {
 namespace native {
 
+template <typename scalar_t>
+void inline flip_cpu_kernel(
+  const int64_t total_dims,
+  const int64_t* stride_contiguous_d,
+  const Tensor& flip_dims_t,
+  const Tensor& in_tensor,
+  Tensor& out_tensor
+){
+  int64_t i;
+  const int64_t numel = in_tensor.numel();
+  const scalar_t* in_tensor_d = in_tensor.data<scalar_t>();
+  scalar_t* out_tensor_d = out_tensor.data<scalar_t>();
+  const int64_t* flip_dims_d = flip_dims_t.data<int64_t>();
+
+  #pragma omp parallel for private(i) if (numel > 1000)
+  for (i = 0; i < numel; i++) {
+    int64_t cur_indices = i;
+    int64_t rem = 0;
+    int64_t dst_offset = 0;
+
+    for (int64_t d = 0; d < total_dims; d++) {
+      int64_t temp = cur_indices;
+      cur_indices = cur_indices / stride_contiguous_d[d];
+      rem = temp - cur_indices * stride_contiguous_d[d];
+      for (int64_t j = 0; j < flip_dims_t.numel(); j++) {
+        if (d == flip_dims_d[j]) cur_indices = in_tensor.size(d) - 1 - cur_indices;
+      }
+      dst_offset += cur_indices * in_tensor.stride(d);
+      cur_indices = rem;
+    }
+    out_tensor_d[i] = in_tensor_d[dst_offset];
+  }
+}
+
 Tensor flip_cpu(const Tensor& self, IntList dims) {
-  const int64_t total_dims = self.dim(), flip_dims_size = dims.size();
-  flip_check_errors(total_dims, flip_dims_size, dims);
+  auto in_tensor = self;
+  const int64_t total_dims = in_tensor.dim();
 
+  dim_list_to_bitset(dims, total_dims); // returned bitset is not used, here only check correctness of dims
```

Comment:
1. does the `maybe_wrap_dims` call not check correctness?
2. I though you were going to make the kernel use the bitset, and save the inner for loop?

Reply:
oops, sorry, I see your point now, yes, the inner loop is not needed